### DISCUSSION TITLE
Ports

### DISCUSSION THREAD

**Discussion by DartPower** [diasurgical/DevilutionX#2450] [Ideas]
Ports for:
PS2
GameCube
Wii
Wii U
Xbox Original
Xbox 360
PS3
Dreamcast
Is Possible? 😄 

or... libretro? 😃

**Comment by AJenbo** [comment]
an Xbox Original port already exists

libretro is just a menu afaik, and people are already using it as a launcher in some cases.

PS2 maybe https://github.com/lukaszdk/ps2doom#readme
GameCube a bit low on memory
Wii maybe https://wiibrew.org/wiki/SDL_Wii
Wii U probably https://github.com/yawut/sdl2-wiiu
Xbox 360 yes https://github.com/LibXenonProject/libSDLXenon
PS3 unclear if other os is required https://github.com/DreamcastSDK/SDL/blob/SDL-1.2/README.PS3
Dreamcast low memory

**Reply by julealgon** [reply]
> GameCube a bit low on memory

oO really? I don't remember the specs, but GC is widely considered a _lot_ more powerful than the PS2 for instance. I can't even imagine such a simple game _not_ running on GameCube.

**Reply by AJenbo** [reply]
It's only got 16MB of ram, PS2 has 32mb, the lowest memory system that we have a full port for is the xbox with 64mb but 32mb should work at this point.

**Reply by julealgon** [reply]
That doesn't seem right. It has more than that according to this:
https://en.wikipedia.org/wiki/GameCube_technical_specifications

> 43 MB total non-unified RAM
> - 24 MB (2x 12 MB) MoSys 1T-SRAM @ 324 MHz (codenamed "Splash") as main system RAM
> - 3 MB embedded 1T-SRAM cache within "Flipper" GPU (2 MB framebuffer/Z-buffer, 1 MB texture cache)
> - 16 MB DRAM used as I/O buffer for audio and DVD drive

**Reply by AJenbo** [reply]
might be possible but will require some work i think.

**Comment by DartPower** [comment]
PSP? :D

**Reply by StephenCWills** [reply]
We have a draft PR for PSP. #1918

**Comment by DartPower** [comment]
How about:
DJGPP (DOS)?
@SerenityOS ?
Sega Saturn ? 
PlayStation 1 ?
😃 😃 😃

**Reply by qndel** [reply]
https://www.youtube.com/watch?v=ZOzZ8R4gphE
looks like there's a serenityos port already
Everything else is no.

**Reply by AJenbo** [reply]
Check the specs: https://github.com/diasurgical/devilutionX/discussions/2450#discussioncomment-1042591

**Comment by DartPower** [comment]
Any news about porting on PS2 and/or Xbox 360? I can test it on real hardware 😄 
P.S. PSP port too i can test on real hw (but i have only a fat - PSP-1000, there memory is smaller than other PSPs)

**Reply by AJenbo** [reply]
PSP: -> https://github.com/diasurgical/devilutionX/discussions/2450#discussioncomment-1248060
(you can subscribe to the PR for notifications about activity)

Someone modded there Xbox 360, haven't heard anything about PS2

**Reply by DartPower** [reply]
Thanks a lot for info!
